terraform/ec2-examples/distributed-ml-training/main.tf

```diff
@@ -0,0 +1,489 @@
+  }
+
+  vpc_zone_identifier = module.vpc.private_subnets
+  health_check_type   = "EC2"
+  min_size            = 2
+  max_size            = 2
+  desired_capacity    = 2
+
+  # https://github.com/hashicorp/terraform-provider-aws/issues/12582
+  autoscaling_group_tags = {
+    AmazonECSManaged = true
+  }
+  block_device_mappings = [
+    {
+      # Root volume
+      device_name = "/dev/xvda"
+      no_device   = 0
+      ebs = {
+        delete_on_termination = true
+        encrypted             = false
+        volume_size           = 50
+        volume_type           = "gp2"
+      }
+    }
+  ]
+
+  tags = local.tags
+}
+
+module "autoscaling_sg" {
+  source  = "terraform-aws-modules/security-group/aws"
+  version = "~> 4.0"
+
+  name                = local.name
+  description         = "Autoscaling group security group"
+  vpc_id              = module.vpc.vpc_id
+  ingress_cidr_blocks = [module.vpc.vpc_cidr_block]
+
+  ingress_rules = ["all-all"]
+
+  egress_rules = ["all-all"]
+
+  tags = local.tags
+}
+
+
+module "ecs_service_head" {
+  source  = "terraform-aws-modules/ecs/aws//modules/service"
+  version = "~> 5.0"
+
+  name               = "distributed_ml_training_head_service"
+  desired_count      = 1
+  cluster_arn        = module.ecs_cluster.arn
+  enable_autoscaling = false
+  memory             = 10240
+  cpu                = 3072
+  # Task Definition
+
+  requires_compatibilities = ["EC2"]
+  capacity_provider_strategy = {
+    default = {
+      capacity_provider = "distributed_ml_training_head" # needs to match name of capacity provider
+      weight            = 1
+      base              = 1
+    }
+  }
+
+  task_exec_iam_role_arn     = aws_iam_role.task_execution_role.arn
+  tasks_iam_role_name        = "taskRole"
+  tasks_iam_role_description = "Task role for ${local.name}"
+  tasks_iam_role_policies = {
+    ReadOnlyAccess = "arn:aws:iam::aws:policy/ReadOnlyAccess"
+  }
+  tasks_iam_role_statements = [
+    {
+      actions = ["s3:*"]
+      resources = [
+        "arn:aws:s3:::${aws_s3_bucket.results.bucket}",
+        "arn:aws:s3:::${aws_s3_bucket.results.bucket}/*"
+      ]
+    }
+  ]
+  create_task_exec_iam_role          = false
+  enable_execute_command             = false
+  deployment_minimum_healthy_percent = 0
+  container_definitions = {
+
+    ray_head = {
+      readonly_root_filesystem = true
+      image                    = local.ray_head_container_image
+      user                     = 1000
+      cpu                      = 3072
+      memory                   = 10240
+      memory_reservation       = 10240
+      command                  = ["/bin/bash", "-lc", "--", "ulimit -n 65536; ray start --head --dashboard-host=0.0.0.0 --metrics-export-port=8080 --num-cpus=0 --memory=10737418240 --block"]
+      linux_parameters = {
+        sharedMemorySize = 20480
+      }
+      mount_points = [{
+        sourceVolume  = "tmp"
+        containerPath = "/tmp"
+        readOnly      = false
+      },
+      {
```

Comment:
during deploying terraform, I faced error: creating ECS Task Definition (distributed_ml_training_worker_service): ClientException: containerPath values aren't unique across mountPoints. 
checking mount_points property is recommended.

Reply:
Fixed